### PR TITLE
fixed bug with ddl board of appeals letters

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- fixed how doctypes were being handled for BOA Letters in DDL
- made it so we can also demo functionality in dev environment using mock data

## Testing done

- updated service spec for RSpec

## What areas of the site does it impact?
DDL

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.

### CODE REVIEW

File: lib/claim_letters/claim_letter_downloader.rb [hunk 1]
```diff
@@ -41,7 +32,7 @@ def get_letter(document_id)
 
       filename = filename_with_date(letter_details[:received_at])
 
-      if !Rails.env.development? && !Rails.env.test?
+      if !Rails.env.development? && !Rails.env.test? && !Settings.vsp_environment == 'development'
```
Comment (kjduensing) [inline lib/claim_letters/claim_letter_downloader.rb:35]: Should this be `Settings.vsp_environment != 'development'`?

Comment (samcoforma) [reply lib/claim_letters/claim_letter_downloader.rb:35]: doh! good catch

**Review by kjduensing — APPROVED** [review]
LGTM

**Review by jerekshoe — APPROVED** [review]

**Review by ryan-mcneil — APPROVED** [review]
👌

**Pull request merged**